acceptance/detector_test.go

```diff
@@ -0,0 +1,209 @@
+// +build acceptance
+
+package acceptance
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	h "github.com/buildpacks/lifecycle/testhelpers"
+
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+)
+
+var (
+	detectDockerContext = filepath.Join("testdata", "detector")
+	detectorBinaryDir    = filepath.Join("testdata", "detector", "container", "cnb", "lifecycle")
+	detectImage         = "lifecycle/acceptance/detector"
+	userID = "1234"
+)
+
+func TestDetector(t *testing.T) {
+	h.SkipIf(t, runtime.GOOS == "windows", "Detector is not yet supported on Windows")
+
+	rand.Seed(time.Now().UTC().UnixNano())
+
+	h.MakeAndCopyLifecycle(t, "linux", detectorBinaryDir)
+	h.DockerBuild(t, detectImage, detectDockerContext)
+	defer h.DockerImageRemove(t, detectImage)
+
+	spec.Run(t, "acceptance-detector", testDetector, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testDetector(t *testing.T, when spec.G, it spec.S) {
+	when("called with arguments", func() {
+		it("errors", func() {
+			cmd := exec.Command("docker", "run", "--rm", detectImage, "some-arg")
+			output, err := cmd.CombinedOutput()
+			h.AssertNotNil(t, err)
+			expected := "failed to parse arguments: received unexpected arguments"
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("running as a root", func() {
+		it("errors", func() {
+			cmd := exec.Command("docker", "run", "--rm", detectImage)
+			output, err := cmd.CombinedOutput()
```

Comment:
I switched to using `h.Run()` and it made the tests terser. Might be worth a try...

Reply:
Oh wait, I see you are using `CombinedOutput()` for precisely the reason I did (assert on error) - nevermind...